File: 백준/Silver/10828. 스택/스택.cc
#include<iostream>
#include <vector>
#include <stack>
#include <algorithm>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int num;
    cin >> num;

    stack<int> st;
    vector <int> vec;

    for (int i = 0; i < num; i++) {
        string a;
        cin >> a;
        if (a == "push") {
            int b;
            cin >> b;
            st.push(b);
        }
        else if (a == "pop") {
            if (st.empty()) vec.push_back(-1);
            else {
                vec.push_back(st.top()); st.pop();
            }
        }
        else if (a == "top") {
            if (st.empty()) vec.push_back(-1);
            else {
                vec.push_back(st.top());
            }
        }
        else if (a == "size") {
            vec.push_back(st.size());

        }
        else if (a == "empty") {
            if (st.empty()) vec.push_back(1);
            else vec.push_back(0);
        }

    }
    for (int res : vec) {
        cout << res << "\n";
    }

    return 0;
}

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.StringTokenizer;
import java.util.Stack;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		Stack<Integer> stack = new Stack<>();
		StringBuilder sb = new StringBuilder();

		int num  = Integer.parseInt(br.readLine());

		for(int i = 0; i < num; i++){
			StringTokenizer st = new StringTokenizer(br.readLine());
			String str = st.nextToken();

			if (str.equals("push")) {
				stack.push(Integer.parseInt(st.nextToken()));
			}
			if (str.equals("pop")) {
				if (stack.isEmpty()) {
					sb.append(-1 + "\n");
				} else {
					sb.append(stack.pop() + "\n");
				}
			}
			if (str.equals("size")) {
				sb.append(stack.size() + "\n");
			}
			if (str.equals("empty")) {
				if (stack.isEmpty()) {
					sb.append(1 + "\n");
				} else {
					sb.append(0 + "\n");
				}
			}
			if (str.equals("top")) {
				if (stack.isEmpty()) {
					sb.append(-1 + "\n");
				} else {
					sb.append(stack.peek() + "\n");
				}
			}
		}
		System.out.println(sb);
	}
}